doc/guides/west/build-flash-debug.rst

```diff
@@ -86,6 +86,18 @@ To use another build directory, use ``--build-dir`` (or ``-d``)::
 
   west build -b <BOARD> --build-dir path/to/build/directory
 
+To change the default build directory from ``build`` to a user-defined format,
+use the ``dir-fmt`` configuration option::
+
+  west config build.dir-fmt "build/{board}/{source_dir}"
+
+With the configuration above, if you build :file:`samples/hello_world` from
+the root of zephyr's repository for the ``reel_board`` the build folder will
+be created at :file:`build/reel_board/samples/hello_world`. If you however
+change your current working directory to :file:`samples/hello_world`, the
+build folder will be created at :file:`build/reel_board`, since the relative
+path to the source directory is just :file:`.`.
```

Comment:
It also would mean the docs here could be shorter, since there's no difference in the results no matter where you run it.

Reply:
Sure, but I wanted to highlight the fact that the tool works both outside and inside the sample folder

Reply:
What I mean is, using {app} leads to uniform results, so there is no need to explain here "if you're here, then it's this, if you're somewhere else, then it's that".

I'm fine with having both.